### PR TITLE
[FEATURE] Add git commit msg hook

### LINKED ISSUE #13: Git commit message hook
Testing commit messages as part of CI is great, but is pretty frustrating for a build to get *that* far and then fail because of something so minor. It's very useful as a final defence to stop bad commits getting through, but it'd be better if there was a client-side hook available in addition to this.

Should create a `commit-msg` hook (and document its usage).

### PR DESCRIPTION
**Pull request by TomSeldon**
Add a commit-msg hook to be used within Git on the client-side.

Closes #13

### CODE REVIEW

File: tools/commit-message-hook.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    if (error) {
+        throw new Error(error);
+    }
+
+    const commitMessage = data.toString();
+    const validationResult = lib.validateCommitMessage(commitMessage);
+
+    if (!validationResult.isValid) {
+        // The commit message is invalid, so output report explaining why.
+        // We could also exit with a non-0 code here, which would block the commit.
```
Comment (rjwaring) [inline tools/commit-message-hook.js]: `// But we haven't, cause we're nice like that.`

File: README.md [hunk 2]
```diff
@@ -122,6 +123,32 @@ Registrations are not currently supported and will be addressed in J#PROJ-988.
...
+
+A `commit-msg` hook is provided which you can setup to run in your development environment, to help catch invalid commit
+messages as they happen.
+
+This is intended to be setup in addition to, and definitely not instead of, checks that run [as part of CI](#as-part-of-ci).
+It's primary purpose is to shorten the feedback time between commiting something, and finding out that the commit
+message is invalid. The alternative being that you don't find out until it fails on CI, which could take a frustrating
+amount of time.
+
+To use the commit hook, copy or symlink the file to `.git/hooks/commit-msg` within your project. i.e. run:
```
Comment (rjwaring) [inline README.md]: Are we assuming that if you want to run multiple things on a particular git hook, that you'll know how to that already?

Comment (TomSeldon) [inline README.md]: I actually have no idea how you would do that. I didn't realise you *could* do that. So no it's definitely not assumed, just overlooked.

Comment (TomSeldon) [inline README.md]: Going to leave this comment as is. It does look like it's possible to have multiple scripts per hook by placing them in `commit-msg.d` folder, but I don't want to stray too much into documenting how Git works (e.g. I don't know if that's true of all version of Git).

**Pull request merged**